staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -666,6 +746,41 @@ func (p *sharedProcessor) resyncCheckPeriodChanged(resyncCheckPeriod time.Durati
 	}
 }
 
+func (p *sharedProcessor) removeListenerFor(handler ResourceEventHandler) {
+	p.listenersLock.Lock()
+	defer p.listenersLock.Unlock()
+
+	listeners := p.removeListenerLockedFor(handler)
+	if p.listenersStarted && len(listeners) > 0 {
```

Comment:
len(listeners) >0 is likely not necessary

Reply:
just changed